.github/actions/maven-build-and-tag/action.yml

```diff
@@ -161,6 +165,14 @@ runs:
         path: |
           **/target/*.jar
 
+    - uses: actions/upload-artifact@v3
+      if: inputs.upload-coverage == 'true'
+      with:
+        name: coverage
+        retention-days: 1
```

Comment:
isn't that too low? I would expect to keep it at least for a week in case you want to compare previous reports?